### PR TITLE
Replace P.clPx with P.clP[x]

### PR DESCRIPTION
**Pull request by zdroid**

### CODE REVIEW

**Comment by matt-gretton-dann** [comment]
This now needs reworking with the changes from #228 merged into master.

**Comment by zdroid** [comment]
@matt-gretton-dann Done.

File: src/CovidSim.cpp [hunk 1]
```diff
@@ -5517,66 +5517,16 @@ int GetInputParameter3(FILE* dat, const char* SItemName, const char* ItemType, v
...
-					else if (n == 2)
-						* ((int*)ItemPtr) = (int)P.clP1;
-					else if (n == 3)
-						sscanf(match, "%s", (char*)ItemPtr);
-				}
-				else if ((match[0] == '#') && (match[1] == '2'))
-				{
+				if (match[0] == '#') {
 					FindFlag++;
+					int match_id = match[1] - '1';
```
Comment (weshinsley) [inline src/CovidSim.cpp:5522]: If you do it this way, should check that match[1] is in range. 
The previous version would have safely ignored anything outside of 1-6, so let's keep it the same.

```
if (match[0] == '#') {
  int match_id = match[1] - '1';
  if ((match_id >= 1) && (match_id <= 6)) {
    FindFlag++;
       ... the if n stuff ...
  }
}
```

Comment (zdroid) [reply src/CovidSim.cpp:5521]: @weshinsley Fixed.

**Pull request merged**